src/bundle/upload.ts

```diff
@@ -301,223 +302,248 @@ async function uploadBundleToCapgoCloud(supabase: SupabaseType, appid: string, b
+    catch (errorUpload) {
+      const endTime = performance.now()
+      const uploadTime = ((endTime - startTime) / 1000).toFixed(2)
+      spinner.stop(`Failed to upload bundle ( after ${uploadTime} seconds)`)
+      p.log.error(`Cannot upload bundle ( try again with --multipart option) ${formatError(errorUpload)}`)
+      if (errorUpload instanceof HTTPError) {
+        const body = await errorUpload.response.text()
+        p.log.error(`Response: ${formatError(body)}`)
+      }
+      // call delete version on path /delete_failed_version to delete the version
+      await deletedFailedVersion(supabase, appid, bundle)
       program.error('')
     }
-    else if (!hasOrganizationPerm(permissions, OrganizationPerm.write)) {
-      p.log.warn('Cannot set channel as a upload organization member')
+
+    const endTime = performance.now()
+    const uploadTime = ((endTime - startTime) / 1000).toFixed(2)
+    spinner.stop(`Bundle Uploaded 💪 (${uploadTime} seconds)`)
+  }
+
+  async function setVersionInChannel(
+    supabase: SupabaseType,
+    options: Options,
+    bundle: string,
+    channel: string,
+    userId: string,
+    orgId: string,
+    appid: string,
+    localConfig: localConfigType,
+    permissions: OrganizationPerm,
+  ) {
+    const { data: versionId } = await supabase
+      .rpc('get_app_versions', { apikey: options.apikey, name_version: bundle, appid })
+      .single()
+
+    if (versionId && hasOrganizationPerm(permissions, OrganizationPerm.write)) {
+      const { error: dbError3, data } = await updateOrCreateChannel(supabase, {
+        name: channel,
+        app_id: appid,
+        created_by: userId,
+        version: versionId,
+        owner_org: orgId,
+      })
+      if (dbError3) {
+        p.log.error(`Cannot set channel, the upload key is not allowed to do that, use the "all" for this. ${formatError(dbError3)}`)
+        program.error('')
+      }
+      const appidWeb = convertAppName(appid)
+      const bundleUrl = `${localConfig.hostWeb}/app/p/${appidWeb}/channel/${data.id}`
+      if (data?.public)
+        p.log.info('Your update is now available in your public channel 🎉')
+      else if (data?.id)
+        p.log.info(`Link device to this bundle to try it: ${bundleUrl}`)
+
+      if (options.bundleUrl) {
+        p.log.info(`Bundle url: ${bundleUrl}`)
+      }
+      else if (!versionId) {
+        p.log.warn('Cannot set bundle with upload key, use key with more rights for that')
+        program.error('')
+      }
+      else if (!hasOrganizationPerm(permissions, OrganizationPerm.write)) {
+        p.log.warn('Cannot set channel as a upload organization member')
+      }
     }
   }
-}
 
-export async function uploadBundle(preAppid: string, options: Options, shouldExit = true) {
-  p.intro(`Uploading`)
-  const pm = getPMAndCommand()
-  await checkLatest()
+  export async function uploadBundle(preAppid: string, options: Options, shouldExit = true) {
```

Comment:
@asr2003 
Before committing, I would recommend you to run `npm run lint`

Reply:
Thanks for the suggestion @ezhil56x